### PR TITLE
Add debugColour and debug switcher to zones

### PR DESCRIPTION
**Pull request by BerkieBb**
Something small that I'd like to add for scripts to do stuff with, for example highlighting a zone.

Usage example:

```lua
local zone = lib.zones.box({...})

-- Maybe later in the code
zone:setDebug(true, vec4(255, 100, 100, 120))
```

### CODE REVIEW

File: imports/zones/client.lua [hunk 1]
```diff
@@ -261,6 +258,21 @@ local function convertToVector(coords)
     return coords
 end
 
+local function setDebug(self, bool, colour)
+    if not bool and insideZones[self.id] then
+        insideZones[self.id] = nil
+    end
+
+    self.debugColour = bool and vec4(colour?.r or self.debugColour?.r or 255, colour?.g or self.debugColour?.g or 42, colour?.b or self.debugColour?.b or 24, colour?.a or self.debugColour?.a or 100) or nil
+    Zones[self.id].debugColour = self.debugColour
```
Comment (thelindat) [inline imports/zones/client.lua:267]: As `Zones[self.id] == self` there shouldn't be a need to reassign this property.

File: imports/zones/client.lua [hunk 2]
```diff
@@ -261,6 +258,21 @@ local function convertToVector(coords)
...
+
+    if bool and self.debug or not bool and not self.debug then return end
+
+    self.triangles = bool and (self.type == 'poly' and getTriangles(self.polygon) or self.type == 'box' and { mat(self.polygon[1], self.polygon[2], self.polygon[3]), mat(self.polygon[1], self.polygon[3], self.polygon[4]) }) or nil
+    self.debug = bool and (self.type == 'sphere' and debugSphere or debugPoly) or nil
+    Zones[self.id].debug = self.debug
```
Comment (thelindat) [inline imports/zones/client.lua:273]: `Zones[self.id] == self`

File: imports/zones/client.lua [hunk 3]
```diff
@@ -328,13 +340,16 @@ lib.zones = {
...
+        data.setDebug = setDebug
 
         if data.debug then
             data.debug = nil
 
             CreateThread(function()
+                data.debugColour = vec4(data.debugColour?.r or 255, data.debugColour?.b or 42, data.debugColour?.g or 24, data.debugColour?.a or 100)
                 data.triangles = getTriangles(data.polygon)
                 data.debug = debugPoly
             end)
```
Comment (thelindat) [inline imports/zones/client.lua:355]: Should probably just call `self:setDebug(...)` to cut down on duplicate code (here and on the other zone types).

File: imports/zones/client.lua [hunk 4]
```diff
@@ -4,6 +4,7 @@ local glm = require 'glm'
 ---@field id number
 ---@field coords vector3
 ---@field distance number
+---@field type string
```
Comment (thelindat) [inline imports/zones/client.lua:7]: String literal type `---@field type 'poly' | 'sphere' | 'box'`.
Might need to mark debug as private and add the setDebug method too.

**Review by thelindat — CHANGES_REQUESTED** [review]

File: imports/zones/client.lua (hunk 1 above)
Comment (BerkieBb) [reply imports/zones/client.lua:267]: Yeah I wasn't sure if it would update it

File: imports/zones/client.lua (hunk 4 above)
Comment (BerkieBb) [reply imports/zones/client.lua:7]: After marking debug as private
image

**Pull request merged**